gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java

```diff
@@ -1140,6 +1164,9 @@ private void validateBuilder(final Builder builder) {
             if (builder.maxWaitForConnection < 1)
                 throw new IllegalArgumentException("maxWaitForConnection must be greater than zero");
 
+            if (builder.connectionSetupTimeoutMillis > builder.maxWaitForConnection)
+                throw new IllegalArgumentException("maxWaitForConnection cannot be less than connectionSetupTimeoutMillis");
```

Comment:
nit: make exception message logic order the same as if check
```suggestion
                throw new IllegalArgumentException("connectionSetupTimeoutMillis cannot be greater than maxWaitForConnection");
```

Reply:
Ack. Leave this as is then.